hudi-common/src/main/java/org/apache/hudi/common/config/DFSPropertiesConfiguration.java

```diff
@@ -117,7 +142,50 @@ public void addProperties(BufferedReader reader) throws IOException {
     }
   }
 
-  public TypedProperties getConfig() {
-    return props;
+  public static TypedProperties getGlobalProps() {
+    final TypedProperties globalProps = new TypedProperties();
+    globalProps.putAll(HUDI_CONF_PROPS);
+    return globalProps;
+  }
+
+  public TypedProperties getProps() {
+    return getProps(false);
+  }
+
+  public TypedProperties getProps(boolean includeHudiConf) {
+    if (includeHudiConf) {
+      TypedProperties mergedProps = new TypedProperties();
+      mergedProps.putAll(HUDI_CONF_PROPS);
+      mergedProps.putAll(props);
+      return mergedProps;
+    } else {
+      return props;
+    }
+  }
+
+  private static Path getDefaultConfPath() {
+    String confDir = System.getenv(CONF_FILE_DIR_ENV_NAME);
+    if (confDir == null) {
```

Comment:
can we have a default for the CONF_FILE_DIR as well? /etc/hudi or something? and fail gracefully if there is no conf there there

Reply:
Good point. I added this logic